lrpar/src/lib/ctbuilder.rs

```diff
@@ -92,67 +93,43 @@ where
 {
 }
 
-#[doc(hidden)]
-#[derive(Clone, PartialEq, Eq, Debug, Hash)]
-pub enum Scope {
-    /// Represents `pub(crate)`
-    Crate,
-    /// Represents `pub(self)`
-    _Self,
-    /// Represents `pub(super)`
-    Super,
-    /// `pub(in String)`
-    In(String),
-}
-
 /// Visibility controls how public generated code is.
 #[derive(Clone, PartialEq, Eq, Debug, Hash)]
 pub enum Visibility {
+    /// No visibiility, just private.
     Private,
-    Public(Option<Scope>),
+
+    /// `pub`
+    Public,
+
+    /// `pub(super)`
+    PublicSuper,
+
+    /// `pub(self)`
+    PublicSelf,
+
+    /// `pub(crate)`
+    PublicCrate,
+
+    /// `pub(in {arg})`
+    PublicIn(String),
 }
 impl Default for Visibility {
     fn default() -> Self {
         Self::Private
     }
 }
-impl Visibility {
-    /// Will generate themodule with the visibility `pub`
-    pub fn public() -> Self {
-        Self::Public(None)
-    }
 
-    /// Will generate the module with the visibilty `pub(super)`
-    pub fn public_super() -> Self {
-        Self::Public(Some(Scope::Super))
-    }
-
-    /// Will generate the module with the visibility `pub(self)`
-    pub fn public_self() -> Self {
-        Self::Public(Some(Scope::_Self))
-    }
-
-    /// Will generate the module with the visibility `pub(crate)`.
-    pub fn public_crate() -> Self {
-        Self::Public(Some(Scope::_Self))
-    }
-
-    /// Will generate the module with the visibility `pub(in {arg})`.
-    pub fn public_in<T: AsRef<str>>(s: T) -> Self {
-        Self::Public(Some(Scope::In(s.as_ref().to_string())))
-    }
-}
-impl fmt::Display for Visibility {
-    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+impl Visibility {
+    /// returns a string to serialize the result
+    pub fn get_str(&self) -> Cow<'static, str> {
```

Comment:
Does having the `Cow` help with the `format!` later? I ask that very naively!

Reply:
Not at all. I just just thinking of saving the extra allocation, it really doesn't matter. I can have it return a `String` if you prefer.

Reply:
If it saves an allocation, I'm all for it: I just couldn't work it out if it did or not. Does it definitely save an allocation on our common path?

Reply:
In the common path? I don't think so. The builder is normally invoked in the `build.rs`.

Reply:
Having thought about it, I see why `Cow` works here. I'm fine with it.